### PR TITLE
Fixed #28160 -- Prevented hiding GDAL exceptions.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/28160

### CODE REVIEW

File: tests/test_runner/test_discover_runner.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def test_testcase_ordering(self):
             # All others can follow in unspecified order, including doctests
             self.assertIn('DocTestCase', [t.__class__.__name__ for t in suite._tests[2:]])
 
-    def test_duplicates_ignored(self):
+    def _test_duplicates_ignored(self):
```
Comment (timgraham) [inline tests/test_runner/test_discover_runner.py]: (to be fixed)

**Comment by claudep** [comment]
I think that `HAS_GEOS` is in the same boat. It may possibly also be removed.

File: tests/test_runner/test_discover_runner.py (hunk 1 above)
Comment (claudep) [inline tests/test_runner/test_discover_runner.py]: We don't have any other cases in the test suite with sub applications?

**Comment by timgraham** [comment]
Okay, I removed `HAS_GEOS` in another commit. The first commit of this PR is in https://github.com/django/django/pull/8461 for easier review.

**Comment by claudep** [comment]
Looks good, thanks. If you have access to a VM without geo libs installed, are you able to test this patch and try to import some `django.contrb.gis` symbols? Hopefully the error messages are clear enough.

**Comment by timgraham** [comment]
The error message is: GDALException: Could not find the GDAL library (tried "gdal", "GDAL", "gdal1.11.0", "gdal1.10.0", "gdal1.9.0", "gdal1.8.0", "gdal1.7.0"). Try setting GDAL_LIBRARY_PATH in your settings.

**Comment by claudep** [comment]
Perfect, then!

**Pull request merged**